tests/abi/tools/revng-check-compatibility-with-abi/Verify.cpp

```diff
@@ -17,52 +17,49 @@ void verifyABI(const TupleTree<model::Binary> &Binary,
                model::ABI::Values ABI);
 
 struct VerificationHelper {
-public:
-  void verify(const abi::FunctionType::Layout &FunctionLayout,
-              const abi::artifact::FunctionArtifact &Artifact) const;
-
-public:
   const model::Architecture::Values Architecture;
   const abi::Definition &ABI;
   const bool IsLittleEndian;
 
 public:
   llvm::StringRef FunctionName = "";
-};
 
-struct IterationAccessHelper {
 public:
-  IterationAccessHelper(const abi::artifact::Iteration &Iteration,
-                        const model::Architecture::Values Architecture) :
-    Iteration(Iteration),
-    RegisterSize(model::Architecture::getPointerSize(Architecture)) {}
-
-  std::optional<llvm::ArrayRef<std::byte>>
-  registerValue(model::Register::Values Register) {
-    auto Iterator = Iteration.Registers.find(Register);
-    if (Iterator != Iteration.Registers.end())
-      return Iterator->Bytes;
-    else
-      return std::nullopt;
-  }
+  using ATest = abi::runtime_test::Named<abi::runtime_test::ArgumentTest>;
+  void arguments(const abi::FunctionType::Layout &, const ATest &) const;
 
-public:
-  const abi::artifact::Iteration &Iteration;
-  const std::size_t RegisterSize;
+  using RVTest = abi::runtime_test::Named<abi::runtime_test::ReturnValueTest>;
+  void returnValue(const abi::FunctionType::Layout &, const RVTest &) const;
+
+private:
+  uint64_t valueFromBytes(llvm::ArrayRef<std::byte> Input) const {
+    std::size_t PointerSize = model::Architecture::getPointerSize(Architecture);
+    revng_assert(Input.size() <= PointerSize);
+
+    uint64_t Result = 0;
+    for (std::size_t I = 0; I < PointerSize; ++I) {
+      if (I != 0)
+        Result <<= 8;
+
+      std::size_t Index = IsLittleEndian ? (PointerSize - I - 1) : I;
+      if (Index < Input.size())
+        Result += static_cast<uint64_t>(Input[Index]);
+    }
+
+    return Result;
+  }
 };
 
 using VH = VerificationHelper;
-void VH::verify(const abi::FunctionType::Layout &FunctionLayout,
-                const abi::artifact::FunctionArtifact &Artifact) const {
-  for (std::size_t Index = 0; Index < Artifact.Iterations.size(); ++Index) {
-    IterationAccessHelper Helper(Artifact.Iterations[Index], Architecture);
-
-    // Sort the argument registers
+void VH::arguments(const abi::FunctionType::Layout &FunctionLayout,
+                   const ATest &Iterations) const {
```

Comment:
What this function does is intuitively clear but it's hard to check for the details.
In substance, it compares two things. Please rename all variables in something like `ExpectedXYZ` and `ActualXYZ` or `LeftXYZ` and `RightXYZ`.

Reply:
Sorry for the wait. While at it I also split it into pieces. Should be a breeze to review it now: please take a look.

Reply:
It looks like you have already taken a look at this (since you closed a related thread). Should this also be closed?